Userland/Applications/Maps/MapWidget.cpp

```diff
@@ -0,0 +1,295 @@
+    }
+}
+
+void MapWidget::paint_scale_line(GUI::Painter& painter, String label, Gfx::IntRect rect)
+{
+    painter.fill_rect(rect, Gfx::Color(255, 255, 255, 204));
+    painter.fill_rect({ rect.x(), rect.y(), 1, rect.height() }, Gfx::Color(51, 51, 51));
+    painter.fill_rect({ rect.x() + rect.width() - 1, rect.y(), 1, rect.height() }, Gfx::Color(51, 51, 51));
+    Gfx::FloatRect label_rect { rect.x() + PANEL_PADDING_X, rect.y() + PANEL_PADDING_Y, rect.width() - PANEL_PADDING_X * 2, rect.height() - PANEL_PADDING_Y * 2 };
+    painter.draw_text(label_rect, label, Gfx::TextAlignment::TopLeft, Gfx::Color(51, 51, 51));
+}
+
+void MapWidget::paint_scale(GUI::Painter& painter)
+{
+    double max_meters = m_center.distance_to({ m_center.latitude, tile_x_to_longitude(longitude_to_tile_x(m_center.longitude, m_zoom) + static_cast<double>(m_scale_max_width) / TILE_SIZE, m_zoom) });
+    float margin_x = 8;
+    float margin_y = 8;
+    float line_height = PANEL_PADDING_Y + painter.font().pixel_size() + PANEL_PADDING_Y;
+
+    // Metric line
+    double meters = nice_round_number(max_meters);
+    float metric_width = m_scale_max_width * (meters / max_meters);
+    Gfx::IntRect metric_rect = { margin_x, height() - margin_y - line_height * 2, metric_width, line_height };
+    if (meters < 1000) {
+        paint_scale_line(painter, MUST(String::formatted("{} m", meters)), metric_rect);
+    } else {
+        paint_scale_line(painter, MUST(String::formatted("{} km", meters / 1000)), metric_rect);
+    }
+
+    // Imperial line
+    double max_feet = max_meters * 3.28084;
+    double feet = nice_round_number(max_feet);
+    double max_miles = max_feet / 5280;
+    double miles = nice_round_number(max_miles);
+    float imperial_width = m_scale_max_width * (feet < 5280 ? feet / max_feet : miles / max_miles);
+    Gfx::IntRect imperial_rect = { margin_x, height() - margin_y - line_height, imperial_width, line_height };
+    if (feet < 5280) {
+        paint_scale_line(painter, MUST(String::formatted("{} ft", feet)), imperial_rect);
+    } else {
+        paint_scale_line(painter, MUST(String::formatted("{} mi", miles)), imperial_rect);
+    }
+
+    // Border between
+    painter.fill_rect({ margin_x, height() - margin_y - line_height, max(metric_width, imperial_width), 1.0f }, Gfx::Color(51, 51, 51));
+}
+
+void MapWidget::paint_attribution(GUI::Painter& painter)
+{
+    m_attribution_width = PANEL_PADDING_X + painter.font().width(m_attribution_text) + PANEL_PADDING_X;
+    m_attribution_height = PANEL_PADDING_Y + painter.font().pixel_size() + PANEL_PADDING_Y;
+    painter.fill_rect({ width() - m_attribution_width, height() - m_attribution_height, m_attribution_width, m_attribution_height }, Gfx::Color(255, 255, 255, 204));
+    Gfx::FloatRect attribution_text_rect { 0.0f, 0.0f, width() - PANEL_PADDING_X, height() - PANEL_PADDING_Y };
+    painter.draw_text(attribution_text_rect, m_attribution_text, Gfx::TextAlignment::BottomRight, Gfx::Color(51, 51, 51));
```

Comment:
These colors are copied from [Leaflet.js](https://leafletjs.com/) and they match well with the currently used OSM map tile images. I would argue to not change these colors for now until we for example other themed map tiles. I see the drawed panels as a part of the complete map widget which uses an other style by default because the OSM tiles don't use the system theme colors.

Reply:
That makes sense, maybe add a comment about this so other contributors won't try to update it? It might help to extract the colors to separate variables as well.